### ISSUE TITLE
Can/will this extension be ported to a "web extension"?

### ISSUE THREAD

**Issue by hwine** [prefiks/u2f4moz#56]
[Mozilla has announced](https://blog.mozilla.org/addons/2016/11/23/add-ons-in-2017/) that only "web extensions" will be usable in Firefox by the end of 2017.

Can this extension be ported?

Will this extension be ported?

**Comment by mk-fg** [comment]
This extension works by running bundled "u2f" binary that communicates with /dev/hidraw0 (or its analogue on windows).
WebExtensions have "Native Messaging" API for this, though I don't think they allow bundling of the binary itself, but it can be downloaded/installed separately, so it looks like straight-up port is at least possible.
There's also chance that by Firefox-57 there will be native u2f support, as it was planned for Q2-3 2017.

**Comment by prefiks** [comment]
Yes that is true, there is no possibility to deliver binary program that is used for communication with u2f device in webextension, to make it work it would require downloading and installing it outside firefox.

**Comment by anarcat** [comment]
i would close this in favor of #59 which has more details, including news from the Mozilla developers that they now include U2F support straight in FF 57!

**Comment by mk-fg** [comment]
@anarcat Thanks for the pointer!
